File: proxy/c-sharp/MaskOrder.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Proxy
{
    // Maske siparişininin oluşturulduğu sınıf.
    // İstemcinin direkt erişmesini istemediğimiz alandır.
    // UML diyagramındaki ConcreteService sınıfına denk gelmektedir.
    class MaskOrder : IOrderableMask
    {
        public void CreateOrder(Person person)
        {
            // Burada siparişin oluşturulma işlemleri yer alacaktır.
            Console.WriteLine($"Order created to {person.Address}.");
        }
    }
}

File: proxy/typescript/mask-order.ts
import {IOrderableMask} from "./orderable-mask.interface";
import {Person} from "./person";

// Maske siparişininin oluşturulduğu sınıf.
// İstemcinin direkt erişmesini istemediğimiz alandır.
// UML diyagramındaki ConcreteService sınıfına denk gelmektedir.
export class MaskOrder implements IOrderableMask{
  public createOrder(person: Person): void {
    // Burada siparişin oluşturulma işlemleri yer alacaktır.
    console.log(`Order created to ${person.address}`);
  }
}
